### PR TITLE
HEVC 8bit vdenc 444 ayuv cqp and gst-msdk cqp encode support

### PR DESCRIPTION
**Pull request by FocusLuo**
Added HEVC 8bit vdenc 444 ayuv cqp and gst-msdk cqp encode support

### CODE REVIEW

File: test/gst-msdk/encode/hevc.py [hunk 1]
```diff
@@ -89,6 +90,37 @@ def test_r2r(self, case, gop, slices, qp, quality, profile):
     vars(self).setdefault("r2r", 5)
     self.encode()
 
+class cqp_lp_444(HEVC8EncoderTest):
```
Comment (uartie) [inline test/gst-msdk/encode/hevc.py:93]: Why do we need a new class for this?  It is the same as the `cqp_lp` class, just different spec and default profiles.  We should use the `cqp_lp` spec and class, instead, for 444 cases.  The profile can be specified in user spec explicitly.

File: test/gst-msdk/encode/encoder.py [hunk 2]
```diff
@@ -121,8 +121,7 @@ def before(self):
     self.refctx = []
 
   def validate_caps(self):
-    # MSDK does not support AYUV input format even though iHD supports it.
-    ifmts = list(set(self.caps["fmts"]) - set(["AYUV"]))
+    ifmts = list(set(self.caps["fmts"]))
```
[the remark this answers is not in the document]
Comment (FocusLuo) [reply test/gst-msdk/encode/encoder.py:124]: Sure, let me update here

File: test/gst-msdk/encode/hevc.py (hunk 1 above)
Comment (FocusLuo) [reply test/gst-msdk/encode/hevc.py:93]: I had wanted to separate the hevc 8bit 444 encode from hevc 8bit encode test result in the test report in future. Just as:
Encode:  hevc
               hevc.lp
               hevc.lp.444
What do you have any suggestion on separate test result in test report?

Comment (uartie) [reply test/gst-msdk/encode/hevc.py:93]: You can filter by profile parameter for now.

In the future, I'd like to add support for user-defined categories that allow users to assign cases to arbitrary categories that can be used to filter sets of cases (e.g. "itu-t", "4k", "allegro", "multires", or any arbitrary user-defined category, etc...).  But we have not designed that feature yet.

Comment (FocusLuo) [reply test/gst-msdk/encode/hevc.py]: Ok, I followed your suggestion and updated the patches

File: test/gst-msdk/encode/encoder.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (FocusLuo) [reply test/gst-msdk/encode/encoder.py]: update done

File: test/gst-msdk/encode/encoder.py [hunk 3]
```diff
@@ -121,8 +121,7 @@ def before(self):
     self.refctx = []
 
   def validate_caps(self):
-    # MSDK does not support AYUV input format even though iHD supports it.
-    ifmts = list(set(self.caps["fmts"]) - set(["AYUV"]))
+    ifmts = set(self.caps["fmts"])
```
Comment (uartie) [inline test/gst-msdk/encode/encoder.py:124]: I don't think you need to cast this to a `set()`

**Pull request merged**